### DISCUSSION TITLE
Packaging quasar based components into a separated package

### DISCUSSION THREAD

**Discussion by kobymol** [quasarframework/quasar#11969] [General - Components / Directives / etc]
We have many different components (custom actions above quasar functionality), all of such are based on quasar and integral of the vue application.

We are would like to start using outsource developers, and we would like them to work on the components but without supplying them the entire application.

Is there a way to gather all of our components as a npm package?

**Comment by yusufkandemir** [comment]
You can use https://github.com/quasarframework/quasar-starter-kit-ui#usage to create a standalone NPM package containing your components, utilities, etc., and another one for an App Extension to integrate the stuff inside your UI package, and do even more stuff like update your quasar and webpack config, add boot files, and lots more. You can follow our [App Extensions documentation](https://quasar.dev/app-extensions/introduction) for more detailed information.

Here are some examples:
- https://github.com/quasarframework/quasar-ui-qcalendar
- https://github.com/quasarframework/quasar-ui-qactivity
- https://github.com/quasarframework/quasar-ui-qmarkdown
- https://github.com/quasarframework/quasar-ui-qmediaplayer
